### PR TITLE
Reminder objectbox

### PR DESCRIPTION
**Pull request by JvnSlv**
#### Overview

> Closes: #674 

#### Checklist

*Implementation*
- [x] Implementation matches ticket acceptance criteria and technical notes
- [x] Manually tested against Acceptance Criteria
- [ ] ~~UI checked in Light / Dark mode~~

*Stability*
- [x] Checked if changes affect any features and verified affected features work as expected
- [ ] ~~Added unit tests for new code and verified existing tests work as expected~~

*Code quality*
- [ ] Updated `CHANGELOG.md`, `README.md` and package versions in `pubspec.yaml`
- [x] Dependencies are updated to latest versions or new tickets are created if there are breaking changes or deprecations
- [ ] If an unrelated part of the codebase needs to be updated or refactored, create tickets with proposed changes

### CODE REVIEW

**Review by DDavidPrime — APPROVED** [review]

**Review by DDavidPrime — APPROVED** [review]

File: examples/reminders/pubspec.yaml [hunk 1]
```diff
@@ -7,44 +7,47 @@ environment:
...
-  flutter_facebook_auth: ^6.0.0
+  flutter_facebook_auth: ^7.0.0
   flutter_localizations:
     sdk: flutter
-  flutter_rx_bloc:
-  flutter_secure_storage: ^8.0.0
-  flutter_slidable: ^3.0.0
-  flutter_sticky_header: ^0.6.0
-  go_router: ^10.0.0
+  flutter_rx_bloc: null
```
Comment (StanevPrime) [inline examples/reminders/pubspec.yaml:23]: Why we do not specify the flutter_rx_bloc version

File: examples/reminders/lib/base/data_sources/local/objectbox_local_data_source.dart [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+      .query(ObjectBoxLocalReminderModel_.complete.equals(false))
+      .build()
+      .find()
+      .length;
+
+  @override
+  Future<ReminderPair> update(ReminderModel updatedModel) async {
+    var oldModel = (await _remindersBox).get(int.parse(updatedModel.id));
+    final newModel = (await _remindersBox).put(
+      ObjectBoxLocalReminderModel(
```
Comment (StanevPrime) [inline examples/reminders/lib/base/data_sources/local/objectbox_local_data_source.dart:109]: From what I see here we have issues with the naming convention and the approach in general.

I believe the `ObjectBoxLocalReminderModel` is not an actual business model, but rather a data transfer object (DTO) used for the object box communication. The same as we use a HashMap in the API communication.

If this is true, please implement the industry established software design pattern described here

DTO (ObjectBox)  -> Mapper -> Business Model
Business Model    -> Mapper -> DTO (ObjectBox)


https://www.baeldung.com/java-dto-pattern

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: examples/reminders/lib/base/data_sources/local/objectbox_local_data_source.dart (hunk 2 above)
Comment (JovanSl) [reply examples/reminders/lib/base/data_sources/local/objectbox_local_data_source.dart]: Done

Comment (JvnSlv) [reply examples/reminders/lib/base/data_sources/local/objectbox_local_data_source.dart]: Done

File: examples/reminders/lib/base/data_sources/local/objectbox_local_data_source.dart [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+      }
+    }
+    return order;
+  }
+}
+
+extension ReminderModelMapper on ReminderModel {
+  ObjectBoxLocalReminderModel toReminderObjectboxLocalModel(
+      int id, String title, DateTime dueDate, bool complete, String? authorId) {
+    return ObjectBoxLocalReminderModel(
```
Comment (StanevPrime) [inline examples/reminders/lib/base/data_sources/local/objectbox_local_data_source.dart:182]: Considering none of the ReminderModel's property is not used what's the point of creating an extension on top of that model ?

File: examples/reminders/lib/base/data_sources/remote/objectbox_remote_data_source.dart [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+      id: id,
+      title: title,
+      dueDate: dueDate,
+      complete: complete,
+      authorId: authorId,
+    );
+  }
+}
+
+extension ReminderRealmModelMapper on ObjectBoxCloudReminderModel {
```
Comment (StanevPrime) [inline examples/reminders/lib/base/data_sources/remote/objectbox_remote_data_source.dart:198]: Do we really need to have a separate data source and DTOs for the cloud ?

File: examples/reminders/lib/base/models/reminder/objectbox_reminder_model.dart [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import 'package:objectbox/objectbox.dart';
+
+@Entity()
+class ObjectBoxLocalReminderModel {
```
Comment (StanevPrime) [inline examples/reminders/lib/base/models/reminder/objectbox_reminder_model.dart:4]: Once again the object box classes should not be part of the business layer

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: examples/reminders/lib/base/data_sources/remote/objectbox_remote_data_source.dart (hunk 4 above)
Comment (JvnSlv) [reply examples/reminders/lib/base/data_sources/remote/objectbox_remote_data_source.dart:198]: Yes, DTOs have different annotations. Cloud uses `@Sync` annotation allowing it to connect to the sync server, but in order to use this DTO, it is required to initialize the sync client.
If we try to use DTO with `@Sync` annotation with local storage, an error is thrown that the sync client is not initialized.

File: examples/reminders/lib/base/data_sources/local/objectbox_local_data_source.dart (hunk 3 above)
Comment (JvnSlv) [reply examples/reminders/lib/base/data_sources/local/objectbox_local_data_source.dart:182]: I'm not sure that I understand this

File: examples/reminders/lib/base/models/reminder/objectbox_reminder_model.dart (hunk 5 above)
Comment (StanevPrime) [reply examples/reminders/lib/base/models/reminder/objectbox_reminder_model.dart:4]: It sounds like a good option to me. @DDavidPrime what do you think?

Comment (DDavidPrime) [reply examples/reminders/lib/base/models/reminder/objectbox_reminder_model.dart:4]: Yeah, `lib/base/dto/*` or maybe `lib/base/models/dto/*` do sound like a good option.

Comment (StanevPrime) [reply examples/reminders/lib/base/models/reminder/objectbox_reminder_model.dart:4]: image

Let's go with the lib/base/dtos option

File: examples/reminders/lib/base/data_sources/local/objectbox_local_data_source.dart (hunk 3 above)
Comment (StanevPrime) [reply examples/reminders/lib/base/data_sources/local/objectbox_local_data_source.dart:182]: image

Make sure you follow the established project structure

File: examples/reminders/lib/base/data_sources/remote/objectbox_remote_data_source.dart [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+  }
+
+  static Future<Box<ObjectBoxCloudReminderDto>> _initCloud(
+      EnvironmentConfig config, Future<Store> openStore) async {
+    var store = await openStore;
+    var reminderBox = store.box<ObjectBoxCloudReminderDto>();
+    //TODO: Add sync client IP
+    final syncServerIp = Platform.isAndroid ? '10.0.2.2' : '127.0.0.1';
+    final syncClient =
+        Sync.client(store, 'ws://$syncServerIp:9999', SyncCredentials.none());
```
Comment (DDavidPrime) [inline examples/reminders/lib/base/data_sources/remote/objectbox_remote_data_source.dart:43]: I guess this comment was forgotten as I see that the code below addresses it.

**Review by DDavidPrime — APPROVED** [review]

**Pull request closed without merging**